os/net/netdev/netdev_getstats.c

```diff
@@ -0,0 +1,91 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#if defined(CONFIG_NET) && CONFIG_NSOCKET_DESCRIPTORS > 0 && defined(CONFIG_NET_NETMON)
+
+#include <errno.h>
+#include <net/lwip/netif.h>
+#include <net/if.h>
+#include "netdev/netdev.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Function: netdev_getstats
+ *
+ * Description:
+ *   Return the specific interface stats
+ *
+ * Parameters:
+ *   arg Type of information to get
+ *
+ * Returned Value:
+ *   0:Success; negated errno on failure
+ *
+ ****************************************************************************/
+
+int netdev_getstats(void *arg)
+{
+	int ret = -EINVAL;
+	struct netif *dev;
+	struct netmon_netdev_stats *stats = (struct netmon_netdev_stats *)arg;
+	if (stats) {
+		char *intf = stats->devname;
+		dev = netdev_findbyname(intf);
+		if (dev) {
+#ifdef CONFIG_NET_STATS
```

Comment:
Why did you surround a part not whole with CONFIG_NET_STATS?
I think this file is written for getting net stats.

Reply:
You are right. I newly added CONFIG_NET_STATS for all stats related functions. Thanks for the comment.